plugin/acl/setup.go

```diff
@@ -0,0 +1,315 @@
+			} else if sourceType == "file" {
+				if !c.NextArg() {
+					return a, c.ArgErr()
+				}
+				rawNetRanges, err = loadNetworksFromLocalFile(c.Val())
+				if err != nil {
+					return a, c.Errf("Unable to load networks from local file: %v", err)
+				}
+			} else {
+				return a, c.Errf("Unexpected token '%s'; expect 'net'", c.Val())
+			}
+
+			if len(rawNetRanges) == 0 {
+				return a, c.Errf("no network is specified")
+			}
+			var sources []net.IPNet
+			for _, rawNet := range rawNetRanges {
+				rawNet = normalize(rawNet)
+				_, source, err := net.ParseCIDR(rawNet)
+				if err != nil {
+					return a, c.Errf("Illegal CIDR notation '%s'", rawNet)
+				}
+				sources = append(sources, *source)
+			}
+			// TODO: do not hard code 'trie' here. (@ihac)
+			p.filter, err = filter.New("trie", sources)
+			if err != nil {
+				return a, c.Errf("Unable to initialize filter: %v", err)
+			}
+			r.Policies = append(r.Policies, p)
+		}
+		a.Rules = append(a.Rules, r)
+	}
+	return a, nil
+}
+
+// normalize appends '/32' for any single ip address.
+func normalize(rawNet string) string {
+	if idx := strings.IndexAny(rawNet, "/"); idx >= 0 {
+		return rawNet
+	}
+	return rawNet + "/32"
+}
+
+// preprocessNetworks replaces all reversed identifier with real subnets or ip addresses.
+func preprocessNetworks(rawNets []string) []string {
+	var nets []string
+	for _, rawNet := range rawNets {
+		switch rawNet {
+		case "PRIVATE":
+			for _, pn := range PrivateNets {
+				nets = append(nets, pn)
+			}
+		case "LOCAL":
+			intfs, err := net.Interfaces()
+			if err != nil {
+				log.Errorf("Failed to get all network interfaces: %v", err)
+				continue
+			}
+			for _, intf := range intfs {
+				addrs, err := intf.Addrs()
+				if err != nil {
+					log.Errorf("Failed to get addresses from interface %s: %v", intf.Name, err)
+					continue
+				}
+				for _, addr := range addrs {
+					n := addr.String()
+					// ":" should be enough to distinguish IPv4 and IPv6.
+					if strings.Contains(n, ":") {
+						continue
+					}
+					nets = append(nets, n)
+				}
+			}
+		case "*":
+			fallthrough
+		case "ANY":
+			return []string{"0.0.0.0/0"}
+		default:
+			nets = append(nets, rawNet)
+		}
+
+	}
+	return nets
+}
+
+func loadNetworksFromLocalFile(fileName string) ([]string, error) {
+	var nets []string
+	file, err := os.Open(fileName)
+	defer file.Close()
+	if err != nil {
+		return nil, err
+	}
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := stripComment(scanner.Text())
+		// skip empty line.
+		if line == "" {
+			continue
+		}
+		nets = append(nets, line)
+	}
+	return nets, nil
+}
+
+// stripComment removes comments.
+func stripComment(line string) string {
+	commentCh := "#"
+	if idx := strings.IndexAny(line, commentCh); idx >= 0 {
+		line = line[:idx]
+	}
+	line = strings.TrimLeftFunc(line, unicode.IsSpace)
+	return strings.TrimRightFunc(line, unicode.IsSpace)
+}
+
+func parseQype(raw string) (uint16, error) {
```

Comment:
this function is not needed because they exist in `miekg/dns`

Reply:
I only see the `TypeToString` in `miekg/dns` but I need to parse string into qtype. Do I miss something?